PixelGamEngineSolution/FlappyBird/Main.cpp

```diff
@@ -1,76 +1,273 @@
+
+		OverSprite = new olc::Sprite("./sprites/gameover.png");
+		OverDecal = new olc::Decal(OverSprite);
+
+		NumbersSprites = {
+			new olc::Sprite("./sprites/0.png"),
+			new olc::Sprite("./sprites/1.png"),
+			new olc::Sprite("./sprites/2.png"),
+			new olc::Sprite("./sprites/3.png"),
+			new olc::Sprite("./sprites/4.png"),
+			new olc::Sprite("./sprites/5.png"),
+			new olc::Sprite("./sprites/6.png"),
+			new olc::Sprite("./sprites/7.png"),
+			new olc::Sprite("./sprites/8.png"),
+			new olc::Sprite("./sprites/9.png"),
+		};
+
+		for (olc::Sprite* InSprite : NumbersSprites)
+		{
+			NumbersDecal.push_back(new olc::Decal(InSprite));
+		}
+
 		return true;
 	}
 
+	// --------------------------------------------------------------------------------
+
 	bool OnUserUpdate(float ElapsedTime) override
 	{
-		// called once per frame
+		if (bResetGame)
+		{
+			ResetGame();
+		}
+
+		Clear(olc::BLACK);
+		DrawDecal(olc::vf2d(0, 0), BgDecal, {0.15f, 0.15f});//Print BackGround
+		SectionThing();
+		DrawBird();
+
+		if (bHasCollided)
+		{
+			DrawDecal(olc::vf2d(ScreenWidth() / 6, ScreenHeight() / 4), OverDecal, { 0.15f, 0.15f });
+			if (GetKey(olc::SPACE).bReleased)
+			{
+				bResetGame = true;
+			}
+		}
+		else
+		{
+			BirdMechanics(ElapsedTime);
+			//screen colittions
+			if (RedBird.GetPosition().y > ScreenHeight() || RedBird.GetPosition().y < -1)
+			{
+				bHasCollided = true;
+			}
+
+			//ui
+			PrintNumber(FlapsCount);
+		}
+		
+		return true;
+	}
+	// --------------------------------------------------------------------------------
 
+	void DrawBird()
+	{
+		if (RedBird.GetVelocityY() > 0)
+		{
+			DrawDecal(RedBird.GetPosition(), UpDecal, { 0.12f, 0.12f });
+		}
+		else
+		{
+			DrawDecal(RedBird.GetPosition(), DownDecal, { 0.12f, 0.12f });
+		}
+	}
+	
+	// --------------------------------------------------------------------------------
 
-		if (GetKey(olc::SPACE).bPressed)
+	void BirdMechanics(const float InElapsedTime)
+	{
+		if (GetKey(olc::SPACE).bPressed && RedBird.GetVelocityY() >= Gravity / 6.0f)
 		{
-			BirdAcceleration_y = 0.0f;
-			BirdVelocity_y = -Gravity / 4.0f;
+			RedBird.Jump(Gravity);
+
+			FlapsCount++;
+			if (FlapsCount > MaxFlapCount)
+			{
+				MaxFlapCount = FlapsCount;
+			}
 		}
 		else
 		{
-			BirdAcceleration_y += Gravity * ElapsedTime;
+			RedBird.Fall(Gravity * InElapsedTime);
 		}
 
-		if (BirdAcceleration_y >= Gravity)
+		if (RedBird.GetAccelerationY() >= Gravity)
 		{
-			BirdAcceleration_y = Gravity;
+			RedBird.SetAccelerationY(Gravity);
 		}
 
-		//physics
-		BirdVelocity_y += BirdAcceleration_y * ElapsedTime;
-		BirdPosition_y += BirdVelocity_y * ElapsedTime;
+		RedBird.UpdatePosition(InElapsedTime);
 
-		int Bird_X = (int)(ScreenWidth() / 3.0f);//Sets the x position of the bird, this never changes
+		LevelPosition += 14.0f * InElapsedTime;
+	}
+	
+	// --------------------------------------------------------------------------------
 
-		//Draw bird
-		if (BirdVelocity_y > 0)
+	void SectionThing()
```

Comment:
maybe you have to change the method name, to something more specific because in this case, you are doing too many things

Reply:
`Section` _can_ be used as a verb, but "sectioning" a "thing" isn't specific at all. I would recommend `SectionScreen` if you really want to continue using it as a verb, but in the end what you're really doing here might be best summed up in `DrawSections` ,  `DrawLevel` or `DrawObstacles`